File: 19. Remove Nth Node From End of List.c
#include<stdlib.h>

struct ListNode {
    int val;
    struct ListNode *next;
};


struct ListNode* removeNthFromEnd(struct ListNode* head, int n) {
    if(head->next == NULL){ 
        head = NULL;
        return head;
    }
    struct ListNode *temp = head;
    struct ListNode *temp1 = temp;
    while(temp1->next != NULL){
        temp1 = temp;
        for(int i=0; i<n; i++){
            temp1 = temp1->next;
        }
        if(temp1 == NULL){
            head = temp->next;
            return head;
        }
        if(temp1->next != NULL){
            temp = temp->next;
        } else{
            temp->next = temp->next->next;
        }
    }
    return head;

}

File: 19-RemoveNthNodeFromEndOfList/19-RemoveNthNodeFromEndOfList.js
// Last updated: 6/22/2026, 12:54:58 PM
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let fast = head, slow = head;

    for(let i=0; i<n; i++) fast = fast.next;

    if(fast === null) {
        head = head.next;
        return head;
    }

    while(fast.next) {
        slow = slow.next;
        fast = fast.next;
    }

    slow.next = slow.next.next;
    
    return head;
};